setup.py

```diff
@@ -8,7 +8,8 @@
 
 def read(*filenames, **kwargs):
     encoding = kwargs.get("encoding", "utf-8")
-    sep = kwargs.get("sep", os.linesep)
```

Comment:
I am not familiar with this file, when will this being executed>?

Reply:
setup.py is used when pip install is called.
extras_require["pre-dev"] will be used with pip install -e ".[pre-dev]"

Reply:
can we verify os.linesep on windows is \n too? I remember some time ago there was a \r\n that crept into it. I could be wrong.

Reply:
This was changed to "\n" specifically as `io.open` in `read(*filenames, **kwargs)` defaults to "\n" as universal line ending no matter on what system. It was technically broken without an issue previously and has been confirmed on Windows.

Reply:
https://docs.python.org/3/glossary.html#term-universal-newlines